Adding initial MSTP changes over PVST code

wajahatrazi opened the pull request:
Currently adding CLI changes in the PVST code for MSTP

jhli-cisco committed:
[cisco|express-boot]: Add support for cisco express boot in sonic-utilities (#3056)

[cisco|express-boot]: Add support for cisco express boot in sonic-utilities

wajahatrazi committed:
Adding initial MSTP changes over PVST code

wajahatrazi committed:
Adding initial MSTP changes over PVST code

wajahatrazi committed:
Adding MSTP CLI

wajahatrazi committed:
Adding MSTP CLI

wajahatrazi committed:
Global MSTP CLI implementation rechecking

wajahatrazi committed:
Global MSTP CLI implementation rechecking

mlok-nokia committed:
[show_tech] Filter out a list of commands in generate_dump script which are not applicable to Supervisor (#3646)

hat I did
On 202205 branch, execute "show techsupport " on Supervisor. It hung in the scripts which call "show queue counters". This PR filters out a list of command which are not applicable to Supervisor. Fixes Nokia-ION/ndk#60

How I did it
Modify generate_dump script by adding code to check if IS_SUPERVISOR, not to executes the "show queue counters".
And also return at the beginning of the following functions which are not applicable on Supervisor:
save_bfd_info()
save_bgp_info()
save_evpn_info()
save_frr_info()
Also fix some of the failure which are shown during the show techsupport.
a) check if file /etc//cron.d/logrote exists before access it in function disable_logrotate() and enable_logrotate()
b) check if directory /var/log/sai_failure_dump/ exists before using this directory.
Add "show reboot-cause history" to the generate_dump
Check if output of "show vrf" is empty, don't access the empty list with "awk" command
How to verify it
"show techsupport" shoudl work fine on Supervisor as well as on LC

Signed-off-by: mlok <marty.lok@nokia.com>

divyachandralekha commented on config/stp.py:
```diff
@@ -331,45 +505,92 @@ def spanning_tree(_db):
+        ctx.fail("MST is already configured")
+
+    if mode == "pvst":
+        disable_global_mst(db)
```
Let's throw a warning or error for any STP mode change, as this action should not be allowed. This approach ensures that users first disable the existing mode before configuring a new one, will better control and avoiding conflicts

divyachandralekha replied:
warning should be like "MSTP is already configured; please disable PVST before enabling MST" and viceversa

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -331,45 +505,92 @@ def spanning_tree(_db):
+
+        fvs = {'mode': mode,
+            'rootguard_timeout': STP_DEFAULT_ROOT_GUARD_TIMEOUT,
+            'forward_delay': STP_DEFAULT_FORWARD_DELAY,
```
MSTP will not have only mode. Rest of the fields are not required

divyachandralekha commented on config/stp.py:
```diff
@@ -331,45 +505,92 @@ def spanning_tree(_db):
+        enable_stp_for_vlans(db) # Enable STP for VLAN by default
+    
+    elif mode == "mst":
+        disable_global_pvst(db)
```
Lets remove this disable api and add error message

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -331,45 +505,92 @@ def spanning_tree(_db):
+               }
+        db.mod_entry('STP', "GLOBAL", fvs)
+
+        bridge_mac = get_bridge_mac_address(db)
```
We don't need to set the default values for STP_MST table

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -331,45 +505,92 @@ def spanning_tree(_db):
+        }
+        db.set_entry('STP_MST', "GLOBAL", mst_fvs)
+
+        do_vlan_to_instance0(db) # VLANs to Instance 0 mapping as part of global configuration
```
do_vlan_to_instance0 - this call is also not required. When MST mode is enabled , STPd will add all VLANs to CIST

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -331,45 +505,92 @@ def spanning_tree(_db):
+        print("PVST is disabled")
+    elif mode == "mst" and current_mode == "mst":
+        disable_global_mst(db)
+        print("MST is disabled")        
```
Please remove the print

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -331,45 +505,92 @@ def spanning_tree(_db):
+
+    if mode == "pvst" and current_mode == "pvst":
+        disable_global_pvst(db)
+        print("PVST is disabled")
```
Please remove this print

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -331,45 +505,92 @@ def spanning_tree(_db):
 
 
 # cmd: STP global root guard timeout
+# MST CONFIGURATION IN THE STP_PORT TABLE
```
Root guard timeout is not valid for MST mode

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -319,6 +423,76 @@ def get_global_stp_priority(db):
+    if param_type not in allowed_params:
+        ctx.fail("Invalid parameter")
+
+    mst_inst_table = db.get_table('STP_MST_INST')
```
This value should be added in STP_MST table (Global). 
MSTP has only one timer which is global

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -391,13 +614,20 @@ def stp_global_forward_delay(_db, forward_delay):
+        db.mod_entry('STP', "GLOBAL", {'forward_delay': forward_delay})
+    elif current_mode == "mst":
+        db.mod_entry('STP_MST', "GLOBAL", {'forward_delay': forward_delay})
+        update_mst_instance_parameters(ctx, db, 'forward_delay', forward_delay)
```
We dnt need this call update_mst_instance_parameters

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -407,27 +637,68 @@ def stp_global_hello_interval(_db, hello_interval):
+        db.mod_entry('STP', "GLOBAL", {'hello_time': hello_interval})
+    elif current_mode == "mst":
+        db.mod_entry('STP_MST', "GLOBAL", {'hello_time': hello_interval})
+        update_mst_instance_parameters(ctx, db, 'hello_time', hello_interval)
```
We dnt need this instance update call

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -407,27 +637,68 @@ def stp_global_hello_interval(_db, hello_interval):
+        db.mod_entry('STP', "GLOBAL", {'max_age': max_age})
+    elif current_mode == "mst":
+        db.mod_entry('STP_MST', "GLOBAL", {'max_age': max_age})
+        update_mst_instance_parameters(ctx, db, 'max_age', max_age)
```
We dnt need this instance update call

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -407,27 +637,68 @@ def stp_global_hello_interval(_db, hello_interval):
+    db.mod_entry('STP', "GLOBAL", {'max_hops': max_hops})
+    if current_mode == "mst":
+        db.mod_entry('STP_MST', "GLOBAL", {'max_hops': max_hops})
+        update_mst_instance_parameters(ctx, db, 'max_hops', max_hops)
```
We dnt need this instance update call

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -407,27 +637,68 @@ def stp_global_hello_interval(_db, hello_interval):
+
+# Bridge priority cannot be set without Instance ID
 # cmd: STP global bridge priority
+# MST CONFIGURATIONS IN THE STP_MST_INST TABLE
```
Global priority set is not applicable for MST

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -441,6 +712,48 @@ def stp_global_priority(_db, priority):
+    ctx = click.get_current_context()
+    db = _db.cfgdb
+    check_if_global_stp_enabled(db, ctx)
+    if revision not in range(MST_MIN_REVISION, MST_MAX_REVISION + 1):
```
Do we need +1?

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -465,6 +778,7 @@ def check_if_stp_enabled_for_vlan(ctx, db, vlan_name):
         ctx.fail("STP is not enabled for VLAN")
 
 
+#config spanning_tree vlan enable <vlan-id>
```
This command is valid only for PVST. Please add check.

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -500,6 +814,7 @@ def stp_vlan_enable(_db, vid):
             db.mod_entry('STP_VLAN_PORT', vlan_intf_key, vlan_intf_entry)
 
 
+# config spanning_tree vlan disable <vlan-id>
```
This command is valid only for PVST. Please add check.

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -512,6 +827,7 @@ def stp_vlan_disable(_db, vid):
     db.mod_entry('STP_VLAN', vlan_name, {'enabled': 'false'})
 
 
+# config spanning_tree vlan forward_delay <vlan-id> <4-30 seconds>
```
This command is valid only for PVST. Please add check.

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -527,7 +843,7 @@ def stp_vlan_forward_delay(_db, vid, forward_delay):
     db.mod_entry('STP_VLAN', vlan_name, {'forward_delay': forward_delay})
 
-
+# config spanning_tree vlan hello <vlan-id> <1-10 seconds>
```
This command is valid only for PVST. Please add check.

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -543,7 +859,7 @@ def stp_vlan_hello_interval(_db, vid, hello_interval):
     db.mod_entry('STP_VLAN', vlan_name, {'hello_time': hello_interval})
 
-
+# config spanning_tree vlan max_age <vlan-id> <6-40 seconds>
```
This command is valid only for PVST. Please add check.

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha commented on config/stp.py:
```diff
@@ -560,6 +876,7 @@ def stp_vlan_max_age(_db, vid, max_age):
     db.mod_entry('STP_VLAN', vlan_name, {'max_age': max_age})
 
 
+# config spanning_tree vlan priority <vlan-id> <0-61440>
```
This command is valid only for PVST. Please add check.

wajahatrazi replied:
I have addressed this comment, please review

divyachandralekha reviewed (commented):
Added review comments. Please check

bingwang-ms committed:
Update PR checker pipeline to point to bookworm (#3651)

* Update PR checker pipeline to point to bookworm

developfast committed:
Update gcu_field_operation_validators.conf.json with Arista-7260CX3-D108C10 (#3650)

wen587 committed:
[config]Restrict YANG validation to Golden Config (#3656)

What I did
The sonic vm image has yang validation issue with default config setup. Limit YANG check to golden config only before fix default factory config.

How I did it
Restrict YANG validation to golden config only.

How to verify it
Unit test and integration test.

xwjiang-ms committed:
Revert "Speed up route_check script" (#3653)

* Revert "Speed up route_check script (#3544)"

This reverts commit 7cbcfda5e517e5c8a8cfe7140468774b107d6280.

* Revert Speed up route_check script

DavidZagury committed:
Fix save_file command in generate_dump (#3647)

* Fix save_file command in generate_dump

* Fix save_file command in generate_dump

kperumalbfn committed:
Update DB migrator script to next branch 202505 (#3664)

Update DB migrator script to next branch 202505

ganglyu committed:
Support empty golden config file for multi-asic (#3670)

What I did
config_sonic_basedon_testbed.yml creates an empty golden_config_db.json file under /etc/sonic for non mx platforms.
When this empty file exist db_migration fails for all asics for non mx platforms.

How I did it
Update db_migrator to support empty golden config file.

How to verify it
Run unit test and end to end test

wajahatrazi committed:
Adding changes suggested in the MSTP code

wajahatrazi committed:
Adding changes suggested in the MSTP code

wajahatrazi committed:
Update MST group commands
